crates/rpc/src/lib.rs

```diff
@@ -71,13 +88,20 @@ pub fn map_requests(req: &RpcRequest) -> Result<Value, RpcErr> {
     }
 }
 
-pub async fn handle_http_request(body: String) -> Json<Value> {
+pub async fn handle_http_request(State(storage): State<Store>, body: String) -> Json<Value> {
     let req: RpcRequest = serde_json::from_str(&body).unwrap();
 
     let res: Result<Value, RpcErr> = match req.method.as_str() {
         "eth_chainId" => client::chain_id(),
         "eth_syncing" => client::syncing(),
-        "eth_getBlockByNumber" => block::get_block_by_number(),
+        "eth_getBlockByNumber" => {
+            let request = GetBlockByNumberRequest::parse(&req.params).unwrap();
```

Comment:
can we return an RpcError here instead of unwrapping?

Reply:
We need to refactor `handle_http_request` to work like `handle_authrpc_request` in order to propagate the error, Ill make a small PR for it

Reply:
Upate: Here it is: #156